scalding-core/src/main/scala/com/twitter/scalding/TupleConverter.scala

```diff
@@ -67,6 +100,23 @@ object TupleConverter extends GeneratedTupleConverters {
   def arity[T](implicit tc: TupleConverter[T]): Int = tc.arity
   def of[T](implicit tc: TupleConverter[T]): TupleConverter[T] = tc
 
+  import shapeless.ops.hlist._
+
+  implicit def hListConverter[H, T <: HList, N <: Nat]
+    (implicit
+     g: TupleGetter[H],
+     len: Length.Aux[H :: T, N],
```

Comment:
I'm still getting a sense of the design here, but any reason you don't constrain the `FromTupleEntry` instances so that the `I` (which could be a type member) matches the length of the `HList`? That should allow you to omit the `N` type parameter here altogether.